src/app/zap-templates/zcl/data-model/chip/time-synchronization-cluster.xml

```diff
@@ -0,0 +1,87 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+
+  <struct name="TimeZoneType">
+    <item name="Offset" type="INT32S" min="-43200" max="50400"/>
+    <item name="ValidAt" type="INT64U"/> 
+    <item name="Name" type="CHAR_STRING" length="64"/>
+  </struct>
+
+  <struct name="DstOffsetType">
+    <item name="Offset" type="INT32S"/>
+    <item name="ValidStarting" type="INT64U"/> 
+    <item name="ValidUntil" type="INT64U"/>
+  </struct>
+
+
+  <enum name="Granularity" type="ENUM8">
+    <item name="NoTimeGranularity" value="0x0"/>
+    <item name="MinutesGranularity" value="0x1"/>
+    <item name="SecondsGranularity" value="0x2"/>
+    <item name="MillisecondsGranularity" value="0x3"/>
+    <item name="MicrosecondsGranularity" value="0x4"/>
+  </enum>
+
+  <enum name="TimeSourceEnum" type="ENUM8">
+    <item name="None" value="0x0"/>
+    <item name="Unknown" value="0x1"/>
+    <item name="Admin" value="0x2"/>
+    <item name="NodeTimeCluster" value="0x3"/>
+    <item name="NonFabricSntp" value="0x4"/>
+    <item name="NonFabricNtp" value="0x5"/>
+    <item name="FabricSntp" value="0x6"/>
+    <item name="FabricNtp" value="0x7"/>
+    <item name="MixedNtp" value="0x8"/>
+    <item name="NonFabricSntpNts" value="0x9"/>
+    <item name="NonFabricNtpNts" value="0xa"/>
+    <item name="FabricSntpNts" value="0xb"/>
+    <item name="FabricNtpNts" value="0xc"/>
+    <item name="MixedNtpNts" value="0xd"/>
+    <item name="CloudSource" value="0xe"/>
+    <item name="Ptp" value="0xf"/>
+    <item name="Gnss" value="0x10"/>
+  </enum>
+
+  <cluster>
+    <domain>General</domain>
+    <name>TimeSync</name>
+    <code>0x0038</code>
+    <define>TIMESYNC_CLUSTER</define>
+    <description>This cluster is used for time synchronization.</description>
+
+    <attribute side="server" code="0x0001" define="UTC_TIME" type="INT64U" writable="false" default="" optional="false">UTCTime</attribute>
+    <attribute side="server" code="0x0002" define="GRANULARITY" type="ENUM8" writable="false" default="NoTimeGranularity" optional="false">Granularity</attribute>
+    <attribute side="server" code="0x0003" define="TIME_SOURCE" type="ENUM8" writable="false" default="None" optional="false">TimeSource</attribute>
+    <attribute side="server" code="0x0004" define="TRUSTED_TIME_NODE_ID" type="NODE_ID" writable="true" default="" optional="false">TrustedTimeNodeId</attribute>
+    <attribute side="server" code="0x0005" define="DEFAULT_NTP" type="CHAR_STRING" writable="true" default="" length="128" optional="false">DefaultNtp</attribute>
+    <attribute side="server" code="0x0006" define="TIME_ZONE" type="ARRAY" entryType="TimeZoneType" writable="true" default="" length="2" optional="false">TimeZone</attribute>
```

Comment:
Array length in XML is in _bytes_.  I doubt you want to allow only 2 bytes here.